core/schema/coremod.go

```diff
@@ -349,9 +417,11 @@ func introspectionRefToTypeDef(introspectionType *introspection.TypeRef, nonNull
 
 	case introspection.TypeKindEnum:
 		return &core.TypeDef{
-			// just call it a string for now
-			Kind:     core.TypeDefKindString,
+			Kind:     core.TypeDefKindEnum,
 			Optional: !nonNull,
+			AsEnum: dagql.NonNull(&core.EnumTypeDef{
+				Name: introspectionType.Name,
+			}),
```

Comment:
Hm, wait, we're deliberately leaving `Values` empty here?

Reply:
Oh that's a good catch

Reply:
Ok so I looked into this one and we're doing exactly the same for Object, we don't need to load the values there since they are loaded from the dagql server itself.

Reply:
Ok so I looked into this one and we're doing exactly the same for Object, we don't need to load the values there since they are loaded from the dagql server itself.